spec/puppet/resource_api/puppet_context_spec.rb

```diff
@@ -19,12 +19,25 @@
       ex
     end
 
-    let(:logging_proxy) { instance_double(Puppet::ResourceApi::PuppetContext::LoggingProxy, 'logging_proxy') }
-
-    it 'calls the Puppet logging infrastructure' do
-      allow(described_class).to receive(:logging_proxy).with(no_args).and_return(logging_proxy)
-      expect(logging_proxy).to receive(:log_exception).with(exception, 'message', trace: true)
+    it 'will log message at error level and with trace' do
+      expect(Puppet::Util::Log).to receive(:create).with(level: :err, message: match('message: x\na\nb\nc'))
       context.log_exception(exception, message: 'message', trace: true)
     end
+
+    it 'will log message at error level and without trace' do
+      expect(Puppet::Util::Log).to receive(:create).with(level: :err, message: match('message'))
+      context.log_exception(exception, message: 'message', trace: false)
+    end
+
+    context 'when Puppet[:trace] is enabled' do
+      before(:each) do
+        Puppet[:trace] = true
```

Comment:
This will poison the settings for everyone else (especially if you also want to test `Puppet[:trace] = false`. Have you tried 
```
allow(Puppet).to recieve[:[].and_call_original
allow(Puppet).to recieve[:[].with(:trace).and_return(true)
```
?